protocol/x/prices/keeper/market_price.go

```diff
@@ -193,3 +193,19 @@ func (k Keeper) GetMarketIdToValidIndexPrice(
 	}
 	return ret
 }
+
+// GetStreamPriceUpdate returns a stream price update from its id.
+func (k Keeper) GetStreamPriceUpdate(
+	ctx sdk.Context,
+	id uint32,
+	snapshot bool,
+) (
+	val types.StreamPriceUpdate,
+) {
+	price, _ := k.GetMarketPrice(ctx, id)
```

Comment:
Log error here or other more graceful error handling? Otherwise `price` is an empty object